README.md
# ✨ Simple Floating Bottom Navigation Bar


Fully customizable Simple floating bottom navigation bar package for flutter.

## 🎮 How To Use

```dart
import 'package:flutter/material.dart';
import 'package:simple_floating_bottom_nav_bar/floating_bottom_nav_bar.dart';
import 'package:simple_floating_bottom_nav_bar/floating_item.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      title: 'Simple Floating NavBar Demo',
      home:  MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  List<FloatingBottomNavItem> bottomNavItems = const [
    FloatingBottomNavItem(
      inactiveIcon: Icon(Icons.home_outlined),
      activeIcon: Icon(Icons.home),
      label: "Home",
    ),
    FloatingBottomNavItem(
      inactiveIcon: Icon(Icons.search_outlined),
      activeIcon: Icon(Icons.search),
      label: "Search",
    ),
    FloatingBottomNavItem(
      inactiveIcon: Icon(Icons.add_circle_outline),
      activeIcon: Icon(Icons.add_circle),
      label: "Add",
    ),
    FloatingBottomNavItem(
      inactiveIcon: Icon(Icons.person_outline),
      activeIcon: Icon(Icons.person),
      label: "Profile",
    ),
  ];

  List<Widget> pages = [
    Container(
      color: Colors.red,
    ),
    Container(
      color: Colors.purple,
    ),
    Container(
      color: Colors.blue,
    ),
    Container(
      color: Colors.yellow,
    ),
  ];
  @override
  Widget build(BuildContext context) {
    return FloatingBottomNavBar(
      pages: pages,
      items: bottomNavItems,
      initialPageIndex: 0,
      backgroundColor: Colors.green,
      bottomPadding: 10,
      elevation: 0,
      radius: 20,
      width: 300,
      height: 65,
    );
  }
}


```


<img width="450" height="900" src="https://raw.githubusercontent.com/Strange-Philip/simple_floating_bottom_nav_bar/main/images/floating.gif">

### ❗️ Note

- Pull requests are welcomed, especially the animations 🙂

## ⭐️ License

MIT License

CHANGELOG.md
## 0.0.3

* TODO: Reamdme updates.


lib/floating_bottom_nav_bar.dart
import 'package:simple_floating_bottom_nav_bar/floating_item.dart';
import 'package:flutter/material.dart';

class FloatingBottomNavBar extends StatefulWidget {
  final List<Widget> pages;
  final int? initialPageIndex;
  final double? elevation;
  final double? radius;
  final double? height;
  final double? width;
  final double? bottomPadding;
  final Color? backgroundColor;
  final List<FloatingBottomNavItem> items;
  final TextStyle? selectedLabelStyle;
  final TextStyle? unselectedLabelStyle;

  const FloatingBottomNavBar(
      {super.key,
      required this.pages,
      this.initialPageIndex,
      this.elevation = 0,
      this.radius = 20,
      this.backgroundColor = Colors.blue,
      this.height = 65,
      this.width = 300,
      this.bottomPadding = 5,
      required this.items,
      this.selectedLabelStyle,
      this.unselectedLabelStyle});

  @override
  State<FloatingBottomNavBar> createState() => _FloatingBottomNavBarState();
}

class _FloatingBottomNavBarState extends State<FloatingBottomNavBar> {
  int selectedIndex = 0;

  void onItemTapped(int index) {
    setState(() {
      selectedIndex = index;
    });
  }

  @override
  void initState() {
    setState(() {
      selectedIndex = widget.initialPageIndex ?? 0;
    });
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: widget.pages.elementAt(selectedIndex),
      ),
      floatingActionButtonLocation: FloatingActionButtonLocation.centerFloat,
      floatingActionButton: Container(
          width: widget.width,
          height: widget.height,
          decoration: BoxDecoration(
            borderRadius: BorderRadius.circular(widget.radius!),
            color: widget.backgroundColor,
          ),
          padding: EdgeInsets.only(bottom: widget.bottomPadding!),
          child: BottomNavigationBar(
            backgroundColor: Colors.transparent,
            fixedColor: Colors.white,
            elevation: widget.elevation,
            type: BottomNavigationBarType.fixed,
            currentIndex: selectedIndex,
            unselectedFontSize: widget.items.any((element) => element.label == null)
                ? 0
                : widget.unselectedLabelStyle?.fontSize ?? 14,
            selectedFontSize: widget.items.any((element) => element.label == null)
                ? 0
                : widget.selectedLabelStyle?.fontSize ?? 14,
            selectedLabelStyle: widget.selectedLabelStyle,
            unselectedLabelStyle: widget.unselectedLabelStyle,
            onTap: onItemTapped,
            items: widget.items
                .map((item) => BottomNavigationBarItem(
                    icon: item.inactiveIcon, activeIcon: item.activeIcon, label: item.label ?? ''))
                .toList(),
          )),
    );
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:simple_floating_bottom_nav_bar/floating_bottom_nav_bar.dart';
import 'package:simple_floating_bottom_nav_bar/floating_item.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      title: 'Simple Floating NavBar Demo',
      home:  MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  List<FloatingBottomNavItem> bottomNavItems = const [
    FloatingBottomNavItem(
      inactiveIcon: Icon(Icons.home_outlined),
      activeIcon: Icon(Icons.home),
      label: "Home",
    ),
    FloatingBottomNavItem(
      inactiveIcon: Icon(Icons.search_outlined),
      activeIcon: Icon(Icons.search),
      label: "Search",
    ),
    FloatingBottomNavItem(
      inactiveIcon: Icon(Icons.add_circle_outline),
      activeIcon: Icon(Icons.add_circle),
      label: "Add",
    ),
    FloatingBottomNavItem(
      inactiveIcon: Icon(Icons.person_outline),
      activeIcon: Icon(Icons.person),
      label: "Profile",
    ),
  ];

  List<Widget> pages = [
    Container(
      color: Colors.red,
    ),
    Container(
      color: Colors.purple,
    ),
    Container(
      color: Colors.blue,
    ),
    Container(
      color: Colors.yellow,
    ),
  ];
  @override
  Widget build(BuildContext context) {
    return FloatingBottomNavBar(
      pages: pages,
      items: bottomNavItems,
      initialPageIndex: 0,
      backgroundColor: Colors.green,
      bottomPadding: 10,
      elevation: 0,
      radius: 20,
      width: 300,
      height: 65,
    );
  }
}
